File: src/main/java/org/jenkinsci/plugins/hw_dsl_stub/HelloWorldBuilder.java
package org.jenkinsci.plugins.hw_dsl_stub;
import hudson.Launcher;
import hudson.Extension;
import hudson.util.FormValidation;
import hudson.model.AbstractBuild;
import hudson.model.BuildListener;
import net.sf.json.JSONObject;
import org.kohsuke.stapler.DataBoundConstructor;
import org.kohsuke.stapler.StaplerRequest;
import org.kohsuke.stapler.QueryParameter;

import javax.servlet.ServletException;
import java.io.IOException;
import org.jenkinsci.plugins.scala.stub.BuilderAdapter;
import org.jenkinsci.plugins.scala.stub.BuildDescriptorAdapter;

public class HelloWorldBuilder extends BuilderAdapter {

    private final String name;

    @DataBoundConstructor
    public HelloWorldBuilder(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public boolean scala_perform(AbstractBuild build, Launcher launcher, BuildListener listener) {
        if (getDescriptor().getUseFrench())
            listener.getLogger().println("Bonjour, "+name+"!");
        else
            listener.getLogger().println("Hello, "+name+"!");
        return true;
    }

    @Override
    public DescriptorImpl getDescriptor() {
        return (DescriptorImpl)super.getDescriptor();
    }

    @Extension // This indicates to Jenkins that this is an implementation of an extension point.
    public static final class DescriptorImpl extends BuildDescriptorAdapter {
        private boolean useFrench;

        public FormValidation doCheckName(@QueryParameter String value)
                throws IOException, ServletException {
            if (value.length() == 0)
                return FormValidation.error("Please set a name");
            if (value.length() < 4)
                return FormValidation.warning("Isn't the name too short?");
            return FormValidation.ok();
        }

        public String getDisplayName() {
            return "Say hello world";
        }

        @Override
        public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
            useFrench = formData.getBoolean("useFrench");

            return super.configure(req,formData);
        }

        public boolean getUseFrench() {
            return useFrench;
        }
    }
}

File: src/main/scala/org/jenkinsci/plugins/hw_scala/HelloWorldBuilder.scala
package org.jenkinsci.plugins.hw_scala

import hudson.model.{BuildListener, FreeStyleBuild}
import hudson.util.FormValidation
import hudson.{Extension, Launcher}
import net.sf.json.JSONObject
import org.jenkinsci.plugins.scala.stub.{BuildDescriptorAdapter, BuilderAdapter}
import org.kohsuke.stapler.{DataBoundConstructor, QueryParameter, StaplerRequest}

import scala.beans.BeanProperty

object HelloWorldBuilder {
  @Extension
  class DescriptorImpl extends BuildDescriptorAdapter {

    @BeanProperty
    var useFrench: Boolean = _

    def doCheckName(@QueryParameter value: String): FormValidation = {
      if (value.length == 0) return FormValidation.error("Please set a name")
      if (value.length < 4) return FormValidation.warning("Isn't the name too short?")
      FormValidation.ok()
    }

    def getDisplayName(): String = "Say hello world"

    override def configure(req: StaplerRequest, formData: JSONObject): Boolean = {
      useFrench = formData.getBoolean("useFrench")
      super.configure(req, formData)
    }
  }
}

class HelloWorldBuilder@DataBoundConstructor
(@BeanProperty val name: String) extends BuilderAdapter {
  override def scala_perform(build: FreeStyleBuild, launcher: Launcher, listener: BuildListener): Boolean = {
    if (getDescriptor.getUseFrench) listener.getLogger.println("Bonjour, " + name + "!") else listener.getLogger.println("Hello, " + name + "!")
    true
  }
  //override def getDescriptor(): DescriptorImpl = {
  //  super.getDescriptor.asInstanceOf[DescriptorImpl]
  //}
  override def getDescriptor(): HelloWorldBuilder.DescriptorImpl = {
    super.getDescriptor.asInstanceOf[HelloWorldBuilder.DescriptorImpl]
  }
}


